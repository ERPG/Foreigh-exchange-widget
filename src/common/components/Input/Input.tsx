import { TextField } from '@material-ui/core';
import React from 'react';

export interface IInputProps {
  value: number;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  label: string;
  name: string;
}

const Input = ({ value, onChange, label, name }: IInputProps) => {
  return (
    <TextField
      color="primary"
      id="currency-amount"
      label={label}
      name={name}
      inputProps={{ inputMode: 'numeric', 'data-fx': 'fx-amount-input' }}
      value={value}
      onChange={onChange}
      variant="standard"
      size={'small'}
    />
  );
};

export default Input;
