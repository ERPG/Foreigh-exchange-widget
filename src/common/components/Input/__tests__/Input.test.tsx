import { cleanup, render, fireEvent } from '@testing-library/react';
import { ChangeEvent } from 'react';
import { customRender } from '../../../../../__tests__/__utils__/testUtils';
import Input from '../Input';

describe('Input component', () => {
  afterEach(cleanup);

  it('Should render component without errors', () => {
    const { container } = render(
      <Input label="" name="test-input" value={0} onChange={() => ''} />
    );

    expect(container).toMatchSnapshot();
  });

  it('Should trigger a change of value when entering data', () => {
    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
      expect(event.target.value).toBe('test');
    };

    const { getByDataFx } = customRender(
      <Input label="" name="test-input" value={0} onChange={handleChange} />
    );

    const inputEl = getByDataFx('fx-amount-input');
    expect(inputEl).toBeDefined();

    fireEvent.change(inputEl, { target: { value: 'test' } });
  });
});
