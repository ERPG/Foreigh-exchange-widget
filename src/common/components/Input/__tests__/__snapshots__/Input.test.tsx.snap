// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should render component without errors 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root"
  >
    
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
        data-fx="fx-amount-input"
        id="currency-amount"
        inputmode="numeric"
        name="test-input"
        type="text"
        value="0"
      />
    </div>
  </div>
</div>
`;
